packages/legend-extension-mapping-generation/src/components/MappingGeneration_LegendStudioPlugin.tsx

```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright (c) 2020-present, Goldman Sachs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import packageJson from '../../package.json';
+import { MappingGenerationEditor } from './MappingGenerationEditor';
+import { MappingGenerationEditorState } from '../stores/MappingGenerationEditorState';
+import {
+  type EditorStore,
+  type DSLMapping_LegendStudioPlugin_Extension,
+  type ModelLoaderExtensionConfiguration,
+  LegendStudioPlugin,
+} from '@finos/legend-studio';
+import type { Entity } from '@finos/legend-model-storage';
+
+const GENERATION_TYPE_NAME = `RELATIONAL_MAPPING_GENERATION`;
+
+export class MappingGeneration_LegendStudioPlugin
+  extends LegendStudioPlugin
+  implements DSLMapping_LegendStudioPlugin_Extension
+{
+  constructor() {
+    super(packageJson.extensions.studioPlugin, packageJson.version);
+  }
+
+  override getExtraModelLoaderExtensionConfigurations(): ModelLoaderExtensionConfiguration[] {
+    return [
+      {
+        modelGenerationConfig: { key: GENERATION_TYPE_NAME },
+        load: async (editorStore: EditorStore): Promise<Entity[]> => {
+          const modelLoaderState = editorStore.modelLoaderState;
+          if (modelLoaderState.modelText.length > 0) {
+            const entities: Entity[] =
+              await editorStore.graphManagerState.graphManager.pureCodeToEntities(
+                modelLoaderState.modelText,
+              );
+            return entities;
```

Comment:
i think here we can do the actual loading logic. You can just call 
editorStore.graphState.loadEntityChangesToGraph(entities, undefined) to load without a hard sdlc save.